File: Completed/Go/71.go
import "strings"

func simplifyPath(path string) string {
    stk := []string{}
    
    pathArr := strings.Split(path, "/")
    for _, elm := range pathArr {
        if elm == "" || elm == "." {
            continue
        } else if elm == ".." {
            if len(stk) > 0 {
                stk = stk[:len(stk)-1]
            }
        } else {
            stk = append(stk, elm)
        }
    }
    
    return "/" + strings.Join(stk, "/")
}

File: Completed/python3/71.py
class Solution:
    def simplifyPath(self, path: str) -> str:
      lst = path.split('/')
      
      result = []
      for tok in lst:
        if tok == '.' or tok == '/' or tok == '':
          continue
        elif tok == '..':
          if len(result) > 0:
            result.pop()
        else:
          result.append(tok)
      return '/' + "/".join(result)
      
        
